### PR TITLE
Adding firebase storage

### PR DESCRIPTION
**Pull request by tjmonsi**
@mbleigh Here's the new PR as requested.

I changed the toString to toGsString because it doesn't show in the api documentation. I think because toString is part of an object's original set of methods so it skips that.

### CODE REVIEW

**Comment by tjmonsi** [comment]
This is connected to #69

**Comment by FluorescentHallucinogen** [comment]
@googlebot I confirm that I'm okay with my commits being contributed to this project.

**Comment by FluorescentHallucinogen** [comment]
@tjmonsi Something's wrong with code changes in this pull request:

screenshot

**Comment by FluorescentHallucinogen** [comment]
@tjmonsi For some reason, the demo doesn't show the storage uri anymore.

**Comment by tjmonsi** [comment]
@FluorescentHallucinogen I don't know why but it says there are no conflicts so I guess it is ok?. Even if I changed it here, it doesn't show that it has changed.

**Comment by FluorescentHallucinogen** [comment]
@tjmonsi You have committed `CRLF` line endings instead of `LF`.

**Comment by tjmonsi** [comment]
@FluorescentHallucinogen damn you're right. Reverted and fixed :)

**Comment by FluorescentHallucinogen** [comment]
@tjmonsi The `storage-uri` property doesn't work. Try https://polymerfire-storage-demo.firebaseapp.com.

File: firebase-messaging.html [hunk 1]
```diff
@@ -39,7 +39,7 @@
 
 ```html
 <firebase-messaging token="{{token}}"></firebase-messaging>
-<firebase-document path="/users/[[user.uid]]/token" data="[[token]]"></firebase-document>
```
Comment (FluorescentHallucinogen) [inline firebase-messaging.html]: Please rebase. See https://github.com/firebase/polymerfire/commit/4640beaf4c726067bbc1de57bfabd6f1bc9b7ad9.

**Comment by tjmonsi** [comment]
@mbleigh is this ok to be merged?

**Comment by FluorescentHallucinogen** [comment]
@mbleigh @e111077 @cdata Please review and merge this pull request.

**Comment by mbleigh** [comment]
Trying to get to it ASAP. I'd like to take one more look, it's just been crazy. Sorry for the delay.

**Comment by christophe-g** [comment]
For info in case this PR is being reviewed, 
There is one sub-PR for this feature here : https://github.com/tjmonsi/polymerfire/pull/6. It is a fix for storage path not being properly set while putting files to storage.

**Comment by christophe-g** [comment]
@mbleigh - any chance this will get merged?
Just asking before release of a new component that will potentially depend on `firebase-storage` (otherwise we will use standard `firebase.storage()` calls). Thanks!

**Comment by stemuk** [comment]
@mbleigh any progress on this one?

**Comment by tjmonsi** [comment]
any news on this?

**Comment by locomuco** [comment]
waiting for this feature way over 1 year now... same question here

**Comment by zho** [comment]
@mbleigh are you still there? hello? @robdodson can you help us?

**Comment by tjmonsi** [comment]
@e111077 alright! Thanks :)

**Pull request merged**